### PR TITLE
Add dismiss on trigger if Sentinel is already shown

### PR DESCRIPTION
**Pull request by ZvonimirMedak**
## Summary

Added a requested feature by the QA team to dismiss Sentinel on the second shake if it has already been presented.

### Type

- [X] **Feature**: This pull request introduces a new feature.
- [ ] **Bug fix**: This pull request fixes a bug.
- [ ] **Refactor**: This pull request refactors existing code.
- [ ] **Documentation**: This pull request updates documentation.
- [ ] **Other**: This pull request makes other changes.

#### Additional information

- [X] This pull request introduces a **breaking change**.

### Description

Added a check in the SourceScreenProvider to see if the SentinelTabBarView is already present, if so, dismiss it. If not, show Sentinel

## Checklist

- [X] I have performed a self-review of my own code.
- [X] I have tested my changes, including edge cases.
- [X] I have added necessary tests for the changes introduced (if applicable).
- [X] I have updated the documentation to reflect my changes (if applicable).

## Additional notes

We can fix the breaking change by implementing a default implementation for it. I would like to hear your opinion on it, I'm more for a default implementation, and this not being a breaking change

### CODE REVIEW

File: Sentinel/Classes/Core/SourceScreenProvider.swift [hunk 1]
```diff
@@ -13,6 +13,12 @@ public protocol SourceScreenProvider {
 
     /// The view controller used for presenting the Sentinel.
     func showTools(for view: some View)
+
+    /// Checks if Sentinel is already shown
+    func shouldShowSentinel() -> Bool
```
Comment (nikolamajcen) [inline Sentinel/Classes/Core/SourceScreenProvider.swift:18]: In this case, based on the description of the method, I would rather go with something like `isSentinelShown`. Also, due to the usage of this, can we remove `Sentinel` from the name?

**Review by nikolamajcen — APPROVED** [review]
Approving this one, still, check the comment 👍
